skbio/diversity/beta/_unifrac.py

```diff
@@ -95,31 +85,30 @@ def unweighted_unifrac(u_counts, v_counts, otu_ids, tree,
        73, 1576-1585 (2007).
 
     .. [3] Lozupone, C., Lladser, M. E., Knights, D., Stombaugh, J. & Knight,
-        R. UniFrac: an effective distance metric for microbial community
-        comparison. ISME J. 5, 169-172 (2011).
+       R. UniFrac: an effective distance metric for microbial community
+       comparison. ISME J. 5, 169-172 (2011).
 
     """
+    normalized = False
+    unweighted = True
     if validate:
-        _validate(u_counts=u_counts, v_counts=v_counts,
-                  otu_ids=otu_ids, tree=tree)
-    u_obs_otu_counts = _observed_otu_counts(u_counts, otu_ids)
-    v_obs_otu_counts = _observed_otu_counts(v_counts, otu_ids)
-    u_obs_nodes = set(tree.observed_node_counts(u_obs_otu_counts))
-    v_obs_nodes = set(tree.observed_node_counts(v_obs_otu_counts))
-    obs_branch_length = sum(o.length or 0.0 for o in u_obs_nodes | v_obs_nodes)
-    if obs_branch_length == 0:
-        # boundary case where both communities have no members
-        return 0.0
-    shared_branch_length = sum(
-        o.length or 0.0 for o in u_obs_nodes & v_obs_nodes)
-    unique_branch_length = obs_branch_length - shared_branch_length
-    unweighted_unifrac = unique_branch_length / obs_branch_length
-    return unweighted_unifrac
+        _validate(u_counts, v_counts, otu_ids, tree)
+
+    boundary = _boundary_case(sum(u_counts), sum(v_counts),
```

Comment:
Can we drop the boundary checks? They're pretty rare, and they complicate the code. If we keep it, let's rename.